### PR TITLE
Migration to SiraUtil 3.0 and BS 1.19

### PR DESCRIPTION
**Pull request by raineaeternal**
As the title says; Migrated from SiraUtil 2.X to SiraUtil 3.0.
* Refactored SiraUtil related code.
* Removed PlaylistManager HarmonyPatching.

### CODE REVIEW

File: AccSaber/Installers/MenuInstaller.cs [hunk 1]
```diff
@@ -1,27 +1,33 @@
 ﻿using AccSaber.Downloaders;
-using AccSaber.HarmonyPatches;
 using AccSaber.Managers;
 using AccSaber.UI.MenuButton;
 using AccSaber.UI.MenuButton.ViewControllers;
 using SiraUtil;
 using Zenject;
+using ASDownloader = AccSaber.Downloaders.AccSaberDownloader;
```
Comment (PulseLane) [inline AccSaber/Installers/MenuInstaller.cs:7]: ?

File: AccSaber/Downloaders/AccSaberDownloader.cs [hunk 2]
```diff
@@ -29,6 +31,12 @@ public async Task<List<AccSaberAPISong>> GetRankedMapsAsync(CancellationToken ca
             return await MakeJsonRequestAsync<List<AccSaberAPISong>>(url, cancellationToken);
         }
 
+        // public async Task<List<AccSaberAPILeaderboard>> GetLeaderboardsAsync(string hash, CancellationToken cancellationToken)
+        // {
+        //     string url = API_URL + LEADERBOARD_ENDPOINT + hash;
+        //     return await MakeJsonRequestAsync<List<AccSaberAPILeaderboard>>(url, cancellationToken);
+        // }
+
```
Comment (PulseLane) [inline AccSaber/Downloaders/AccSaberDownloader.cs]: nitpick, but this (along with the LEADERBOARD_ENDPOINT) didn't need to be committed here.

**Review by PulseLane — APPROVED** [review]

File: AccSaber/Downloaders/AccSaberDownloader.cs (hunk 2 above)
Comment (raineaeternal) [reply AccSaber/Downloaders/AccSaberDownloader.cs]: I forgor :skull: - Will remove it all together in another commit.

File: AccSaber/Installers/MenuInstaller.cs (hunk 1 above)
Comment (raineaeternal) [reply AccSaber/Installers/MenuInstaller.cs:7]: No clue what happened here lol

File: AccSaber/Downloaders/AccSaberDownloader.cs (hunk 2 above)
Comment (raineaeternal) [reply AccSaber/Downloaders/AccSaberDownloader.cs]: Changed in ab9224d

**Pull request merged**